vignettes/custom-steps.Rmd

```diff
@@ -33,7 +33,12 @@ We recommend following the same pattern for your custom steps.
 There should be no need to write steps on your own. 
 However, if you have the need for a repeated use of specific combinations of `add_code_step()`, please let us know so in the repo by opening an [issue](https://github.com/ropenscilabs/tic/issues) so that we can discuss whether it makes sense to implement it as a custom step.
 
-In the following, we describe the three methods which the new step class must override:
+## Public methods of the `TicStep` class
+
+The `TicStep` class is the main class that is inherited by all `step_xxx` functions.
+It has three public methods.
+Two of them (`prepare()` and `run()`) must be overriden by the new step class.
```

Comment:
`run()` needs to be overridden so that the step has an effect, the other two have suitable implementations in the base class which kick in when the step doesn't override.